File: server/restaurant/src/main/java/hu/elte/alkfejl/restaurant/entity/response/ReviewsResponse.java
package hu.elte.alkfejl.restaurant.entity.response;

import hu.elte.alkfejl.restaurant.entity.Product;
import lombok.Data;
import lombok.EqualsAndHashCode;

import java.sql.Timestamp;
import java.util.List;

@Data
@EqualsAndHashCode
public class ReviewsResponse {

    private Product product;

    private List<ReviewResponse> reviews;

    @Data
    @EqualsAndHashCode(callSuper = true)
    public static class ReviewResponse extends BaseResponse {

        private Timestamp createDate;

        private Byte stars;

        private String description;

        private String userFirstName;

        private String userLastName;
    }
}

File: client/restaurant/src/app/models/responses/reviews-response.ts
import { BaseModel } from '../base-model';
import { Product } from '../product';

export class ReviewsResponse {

    public product: Product;
    public reviews: ReviewResponse[];

    constructor(object) {
        Object.assign(this, object);
    }
}

class ReviewResponse extends BaseModel {

    public createDate: number;
    public stars: number;
    public description: string;
    public userFirstName: string;
    public userLastName: string;

    constructor(object) {
        super(object.id);
        Object.assign(this, object);
    }
}
